File: data/max_sq_area/Original_Solution.java
class Solution {
  private static final int modulo = (int) 1e9 + 7;

  public int maximizeSquareArea(int m, int n, int[] hFences, int[] vFences) {
    Set<Integer> seen = new HashSet<>();
    int l = hFences.length + 2;
    int[] h = Arrays.copyOf(hFences, l);
    // add both boundary
    h[l - 2] = 1;
    h[l - 1] = m;
    for (int i = 0; i < l; i++) {
      for (int j = i + 1; j < l; j++) {
        seen.add(Math.abs(h[j] - h[i]));
      }
    }

    l = vFences.length + 2;
    int[] v = Arrays.copyOf(vFences, l);
    // add both boundary
    v[l - 2] = 1;
    v[l - 1] = n;

    int gap = 0;
    // check vertical fences
    for (int i = 0; i < l; i++) {
      for (int j = i + 1; j < l; j++) {
        int distance = Math.abs(v[j] - v[i]);
        if (seen.contains(distance)) {
          gap = Math.max(gap, distance);
        }
      }
    }

    if (gap == 0) {
      return -1;
    }

    return (int) ((long) gap * gap % modulo);
  }
}

File: data/max_sq_area/LeetCode_Solutions/Original_Solution.py

class Solution:
    def maximizeSquareArea(self, m: int, n: int, hFences: List[int], vFences: List[int]) -> int:
        sects = defaultdict(list)
        hFences.insert(0, 1); hFences.append(m)
        vFences.insert(0, 1); vFences.append(n)
        hFences.sort(); vFences.sort()
        answer = -1
        for h in hFences:
            for v in vFences:
                if h-v not in sects:
                    sects[h-v] = [h, v]
                else:
                    answer = max(answer, (h-sects[h-v][0]) * (v-sects[h-v][1]))
        if answer == -1: return -1
        return answer % 1000000007    